packages/block-editor/src/components/block-list/block.js

```diff
@@ -123,7 +132,13 @@ function BlockListBlock( {
 
 	// For aligned blocks, provide a wrapper element so the block can be
 	// positioned relative to the block column.
-	if ( isAligned ) {
+	// This is only kept for classic themes that don't support layout
+	// Historically we used to rely on extra divs and data-align to
+	// provide the alignments styles in the editor.
+	// Due to the differences between frontend and backend, we migrated
+	// to the layout feature, and we're now aligning the markup of frontend
+	// and backend.
+	if ( isAligned && ! themeSupportsLayout ) {
```

Comment:
Just an idea: is there any way we could move this code into a filter and together with the code you've added in #38657 that also restores the div wrapper on the front end if the theme does not support layout?

Reply:
I think it'd add a bit of code duplication to do so, and it's not clear what value this would bring. 
Also, the filter for the frontend is in php, the one for the editor need to be in JS.

Reply:
To expand a bit here:

 - we still need to support `getEditWrapperProps` before checking the alignment.
 - we still need to run after `editor.BlockListBlock` filter as this could also inject the alignment.

The above makes it very hard to do in a filter and would create some unnecessary breaking changes.